docs/terms/profiler.md

```diff
@@ -1,15 +1,163 @@
+
+## Features
+
+### Multiple types of Profilers available
+
+There are multiple types of Profilers built in to Great Expectations.  Below is a list with overviews of each one.  For more information, you can view their docstrings and source code in the `great_expectations\profile` [folder on our GitHub](https://github.com/great-expectations/great_expectations/tree/develop/great_expectations/profile).
+
+#### UserConfigurableProfiler
+
+The `UserConfigurableProfiler` is used to build an Expectation Suite from a Dataset. The Expectations built are strict - they can be used to determine whether two tables are the same.
+
+#### BasicDatasetProfiler
+
+The `BasicDatasetProfiler` examines a Batch of data and creates a report that answers the basic questions most data practitioners would ask about a dataset during exploratory data analysis. The Profiler reports how unique the values in the column are, as well as the percentage of empty values in it. Based on the column's type it provides a description of the column by computing a number of statistics, such as min, max, mean and median, for numeric columns, and distribution of values, when appropriate.
+
+#### BasicSuiteBuilderProfiler
+
+The `BasicSuiteBuilderProfiler` is an extension of the `BasicDatasetProfiler` that creates a set of coarse Expectations for the profiled dataset.  Configuration of this Profiler is optional; if no configuration is provided, this Profiler will create Expectations for every column in the dataset.  
+
+The values in Expectations created by the `BasicSuiteBuilderProfiler` are extremely coarse and do not correspond to the actual values in the profiled data.  Rather, the purpose of this Profiler is to generate a broad range of candidate Expectations for the profiled data, which you can then prune down to those Expectations you find relevant and edit with values based on domain knowledge.  You will definitely want to edit the values used in Expectations created by this Profiler! 
+
+#### ColumnsExistProfiler
+
+The `ColumnsExistProfiler` is an extension of the `BasicDatasetProfiler` that additionally adds an Expectation that each column found in the profiled dataset exists in the source data system. 
+
+#### JsonSchemaProfiler
+
+The `JsonSchemaProfiler` creates Expectation Suites from JSONSchema artifacts. Basic suites can be created from these specifications.
+
+:::note
+- There is not yet a notion of nested data types in Great Expectations so suites generated by a `JsonSchemaProfiler` use column map expectations.
+- A `JsonSchemaProfiler` does not traverse nested schemas and requires a top level object of type `object`.
+:::
+
+For an example of how to use the `JsonSchemaProfiler`, see our guide on [how to create a new Expectation Suite by profiling from a JsonSchema file](../guides/expectations/advanced/how_to_create_a_new_expectation_suite_by_profiling_from_a_jsonschema_file.md).
+
+### Extendable to account for domain knowledge
```

Comment:
This is something that afaik is exclusive to Rule Based Profilers. You could technically subclass anything else and do anything, but there is a direct path to doing this with RBPs.